### ISSUE TITLE
warning for unbalanced hyper2 objects

### ISSUE THREAD

**Issue by RobinHankin** [RobinHankin/hyper2#102]
If the fillup value is on the same footing as the other strengths, the sum of the powers should be zero and this is sometimes called a "balanced" likelihood function  If a likelihood function is not balanced, this is usually because of a mistake of some kind.

My first thought was to have a check in the constructor for non-balanced powers.  But then the following idiom:

```
icons["L"] %<>% inc()
icons[c("L","NB")] %<>% dec()
```

(corresponding to a person choosing `L` from a field of `L` and `NB`) would fail on the first line, the temporary object being unbalanced.  Maybe it would be nice to have a (user-controllable?) feature in the print method that gives a massive warning if the `hyper2` object is not balanced.

**Comment by RobinHankin** [comment]
But just including a line 

`if(sum(powers) != 0) {warning("powers not balanced!")}`

in `print.hyper2()` is not a perfect solution.  Currently:

```
> chess["Anand"]
log(Anand^36)
>
```

and this nicely shows that Anand won a total of 36 games.  But with the warning in place:

```
> chess["Anand"]
log(Anand^36)
Warning message:
In print.hyper2(x) : powers not balanced!
> 
```

. . . and I am not sure that this warning is warranted in this case.  The problem comes if we try do _do_ something with `chess["Anand"]`.  Assigning this to a variable would be problematic, for example, as would something like `maxp()`:

```
> maxp(chess["Anand"])
 Topalov    Anand   Karpov 
0.000001 0.999998 0.000001 
> 
```

above, see how the non-balanced nature of the argument to `maxp()` causes the evaluate to effectively be pure Anand.  Maybe `maxp()` should check that its argument is balanced?  Or maybe `chess["Anand"]` should not be a `hyper2` object?

**Comment by RobinHankin** [comment]
fixed in e10a165
